pkg/gce-pd-csi-driver/controller.go

```diff
@@ -2165,13 +2165,21 @@ func pickZonesFromTopology(top *csi.TopologyRequirement, numZones int, locationT
 			}
 
 			// For zonal -> regional disk cloning, one of the replicated zones must match the source zone.
-			if locationTopReq.srcReplicationType == replicationTypeNone {
+			if !locationTopReq.srcIsRegional {
```

Comment:
nit: I think this is more readable as "locationTopReq.srcReplicationType == replicationTypeNone" because the comment says for  zonal -> regional cloning... You are changing this to be for !zonal -> regional cloning, which is just extra logic to do in your head.